pkg/sql/schemachanger/scplan/internal/rules/dep_index_and_column.go

```diff
@@ -670,8 +649,24 @@ func init() {
 				currentStatus(to.node, scpb.Status_PUBLIC),
 			}
 		})
+	// We need to make sure that no columns are added to the index after it
+	// receives any data due to a backfill.
+	registerDepRule("index-column added to index before index is backfilled",
+		scgraph.Precedence,
+		"index", "index-column",
+		func(from, to nodeVars) rel.Clauses {
+			return rel.Clauses{
+				from.el.Type((*scpb.IndexColumn)(nil)),
+				to.el.Type((*scpb.PrimaryIndex)(nil), (*scpb.SecondaryIndex)(nil)),
+
+				joinOnIndexID(from.el, to.el, "table-id", "index-id"),
+				targetStatusEq(from.target, to.target, scpb.ToPublic),
+				currentStatus(to.node, scpb.Status_PUBLIC),
+				currentStatus(from.node, scpb.Status_BACKFILLED),
```

Comment:
Shouldn't this be `Status_BACKFILL_ONLY`? Also, arent' `from` and `to` mixed up? If so I doubt that this rule ever actually does anything.

Reply:
Perhaps we should have tests that check the rule definitions for node bindings which aren't ever generated by opgen.